### PR TITLE
Regmapping fix rocket

### PR DESCRIPTION
**Pull request by depappas-sifive**
This PR does the following:

- removes a level in the reg mapping annotation cases classes
- removes the register name from each reg field
-adds a device name to the register

This change only affects the new reg mapping JSON files.

### CODE REVIEW

**Comment by depappas-sifive** [comment]
only the annotation-generated ones

File: src/main/scala/util/Annotations.scala [hunk 1]
```diff
@@ -142,13 +142,18 @@ object GenRegDescsAnno {
...
     val anonRegFieldName = s"unnamedRegField${byteOffset.toHexString}_${bitOffset}"
     val selectedRegFieldName = regField.desc.map(_.name).getOrElse(anonRegFieldName)
 
     val map = Map[BigInt, (String, String)]() // TODO
 
+//    ("enumerations" -> desc.map {d =>
+//      Option(d.enumerations.map { case (key, (name, edesc)) =>
+//        (("value" -> key) ~ ("name" -> name) ~ ("description" -> edesc))
+//      }).filter(_.nonEmpty)}) )
+
```
[the remark this answers is not in the document]
Comment (mwachs5) [reply src/main/scala/util/Annotations.scala]: enumerations are emitted as part of the "old" JSON. I don't know what your plan is for handling them in the "new" JSON

**Comment by jackkoenig** [comment]
Bumping Chisel (and FIRRTL for that matter) should be fine, do y'all want to do it here or I can make a PR?

**Comment by depappas-sifive** [comment]
PR

**Comment by depappas-sifive** [comment]
I was planning on handling enumerations and the get me my module hack in a
separate PR

File: src/main/scala/util/Annotations.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (depappas-sifive) [reply src/main/scala/util/Annotations.scala]: The commented out enumerations was copied form Megan's JSON writer in RegRouter.

I will handle enums in a separate PR.

**Pull request closed without merging**